<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>995</width>
    <height>633</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>智能安防集成软件</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>推理</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_4" rowstretch="1,1,15" columnstretch="1,1,3">
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="2">
           <widget class="QLabel" name="label_2">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="text">
             <string>图像显示窗口</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="Imglabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <kerning>true</kerning>
             </font>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QSplitter" name="splitter">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="autoFillBackground">
               <bool>true</bool>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="handleWidth">
               <number>5</number>
              </property>
              <widget class="QPushButton" name="starButton">
               <property name="text">
                <string>模型初始化</string>
               </property>
              </widget>
              <widget class="QPushButton" name="cameraButton">
               <property name="text">
                <string>打开摄像头</string>
               </property>
              </widget>
              <widget class="QPushButton" name="OpenButton">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>导入视频/图片</string>
               </property>
              </widget>
              <widget class="QPushButton" name="ReButton">
               <property name="text">
                <string>重新检测</string>
               </property>
              </widget>
              <widget class="QPushButton" name="ContinueButton">
               <property name="text">
                <string>继续检测</string>
               </property>
              </widget>
              <widget class="QPushButton" name="PauseButton">
               <property name="text">
                <string>暂停检测</string>
               </property>
              </widget>
              <widget class="QPushButton" name="stopButton">
               <property name="text">
                <string>结束检测</string>
               </property>
              </widget>
              <widget class="QPushButton" name="EndButton">
               <property name="text">
                <string>退出程序</string>
               </property>
              </widget>
              <widget class="QPushButton" name="weightButton">
               <property name="text">
                <string>请选择用于推理的权重（默认yolov8.pt）</string>
               </property>
              </widget>
              <widget class="QCheckBox" name="checkBox">
               <property name="text">
                <string>是否保存</string>
               </property>
              </widget>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>结果显示窗口</string>
               </property>
              </widget>
              <widget class="QTextEdit" name="textEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="lineWrapColumnOrWidth">
                <number>0</number>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="text">
             <string>设置</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="Line" name="line_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>标注</string>
       </attribute>
       <widget class="QPushButton" name="labelButton">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>390</y>
          <width>161</width>
          <height>71</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>13</pointsize>
         </font>
        </property>
        <property name="text">
         <string>点击开始执行标注</string>
        </property>
       </widget>
       <widget class="QTextEdit" name="classEdit">
        <property name="geometry">
         <rect>
          <x>570</x>
          <y>70</y>
          <width>361</width>
          <height>441</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>590</x>
          <y>20</y>
          <width>321</width>
          <height>31</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>请在此编辑预标注文件（每一行写一个类别）</string>
        </property>
       </widget>
       <widget class="QPushButton" name="safeButton">
        <property name="geometry">
         <rect>
          <x>720</x>
          <y>540</y>
          <width>75</width>
          <height>23</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>保存</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>340</y>
          <width>511</width>
          <height>31</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>提示：txt标注文件请放在图片文件夹的同级目录labels中（点击自动生成）</string>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>70</y>
          <width>314</width>
          <height>51</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="labelbutton2">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>点击选择需要标注的图片文件夹</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>200</x>
          <y>220</y>
          <width>195</width>
          <height>21</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <pointsize>11</pointsize>
            </font>
           </property>
           <property name="text">
            <string>请选择目标类别的个数</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox">
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>11</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>12</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>13</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>14</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>15</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>16</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>17</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>18</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>19</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>20</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>训练</string>
       </attribute>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>470</x>
          <y>120</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>13</pointsize>
         </font>
        </property>
        <property name="text">
         <string>模型参数配置</string>
        </property>
       </widget>
       <widget class="Line" name="line_5">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>100</y>
          <width>1111</width>
          <height>16</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBox_2">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>320</y>
          <width>161</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>是否保存训练过程及权重</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget_2">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>370</y>
          <width>201</width>
          <height>21</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="font">
            <font>
             <pointsize>11</pointsize>
            </font>
           </property>
           <property name="text">
            <string>请选择训练优化器</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="optimizer_comboBox">
           <item>
            <property name="text">
             <string>SGD</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Adam</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>AdamW</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>RMSProP</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>440</y>
          <width>221</width>
          <height>19</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>请选择初始化模型</string>
        </property>
       </widget>
       <widget class="QPushButton" name="train_Button">
        <property name="geometry">
         <rect>
          <x>660</x>
          <y>330</y>
          <width>191</width>
          <height>81</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>15</pointsize>
         </font>
        </property>
        <property name="text">
         <string>开始训练</string>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>170</y>
          <width>361</width>
          <height>121</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="1" colspan="2">
          <widget class="QLabel" name="batch_label">
           <property name="text">
            <string>训练的批次大小：16</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="4" column="1" colspan="2">
          <widget class="QLabel" name="workers_label">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>载入数据的进程数：0</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSlider" name="epoch_Slider">
           <property name="minimum">
            <number>10</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
           <property name="sliderPosition">
            <number>100</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksAbove</enum>
           </property>
           <property name="tickInterval">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QSlider" name="batch_Slider">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>256</number>
           </property>
           <property name="singleStep">
            <number>2</number>
           </property>
           <property name="pageStep">
            <number>10</number>
           </property>
           <property name="sliderPosition">
            <number>16</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksAbove</enum>
           </property>
           <property name="tickInterval">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1" colspan="2">
          <widget class="QLabel" name="epoch_label">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>训练的回合数：100</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QSlider" name="workers_Slider">
           <property name="maximum">
            <number>8</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksAbove</enum>
           </property>
           <property name="tickInterval">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>20</y>
          <width>291</width>
          <height>54</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QPushButton" name="pushButton">
           <property name="text">
            <string>选择图片所在目录</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_2">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QPushButton" name="splitButton">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>一键划分数据集</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QComboBox" name="model_comboBox">
        <property name="geometry">
         <rect>
          <x>240</x>
          <y>440</y>
          <width>121</width>
          <height>19</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>yolov8n.pt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>yolov8n.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>yolov8n-cls.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>yolov8n-seg.yaml</string>
         </property>
        </item>
       </widget>
       <widget class="QComboBox" name="yaml_comboBox">
        <property name="geometry">
         <rect>
          <x>757</x>
          <y>171</y>
          <width>151</width>
          <height>19</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>自定义配置文件</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Argoverse.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco8.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco8-pose.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco8-seg.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco128.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco128-seg.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>coco-pose.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GlobalWheat2020.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ImageNet.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Objects365.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>VisDrone.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>SKU-110K.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>VOC.yaml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>xView.yaml</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>571</x>
          <y>171</y>
          <width>180</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>训练数据集的配置yaml文件</string>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>560</x>
          <y>30</y>
          <width>386</width>
          <height>44</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>    训练需要yaml文件,若无现成的yaml文件可在此处临时创建一个</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="yamlButton">
           <property name="text">
            <string>点击创建所选数据集的yaml文件</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>995</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
